beacon_node/store/src/hot_cold_store.rs

```diff
@@ -2670,10 +2670,10 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
                 ops.push(StoreOp::DeleteExecutionPayload(block_root));
             }
 
-            if slot == anchor_slot {
+            if slot < anchor_info.oldest_block_slot {
```

Comment:
I think we want to stop as soon as we hit the oldest block slot, i.e. `==`

```suggestion
            if slot <= anchor_info.oldest_block_slot {
```